networkx/generators/tests/test_directed.py

```diff
@@ -96,22 +105,26 @@ class TestRandomKOutGraph:
 
     """
 
-    def test_regularity(self):
+    @pytest.mark.parametrize("use_numpy", [False] + ([True] if has_numpy else []))
+    def test_regularity(self, use_numpy):
         """Tests that the generated graph is `k`-out-regular."""
+        fun = _random_k_out_graph_numpy if use_numpy else _random_k_out_graph_python
         n = 10
         k = 3
         alpha = 1
-        G = random_k_out_graph(n, k, alpha)
+        G = fun(n, k, alpha)
         assert all(d == k for v, d in G.out_degree())
-        G = random_k_out_graph(n, k, alpha, seed=42)
+        G = fun(n, k, alpha, seed=42)
```

Comment:
This is completely fine, though the logic in the parametrization is a bit tricky. An alternative that came to mind:

```python
    @pytest.mark.parametrize("f", (_random_k_out_graph_numpy, _random_k_out_graph_python))
    def test_regularity(self, f):                                               
        """Tests that the generated graph is `k`-out-regular."""                
        if (f == _random_k_out_graph_numpy) and not has_numpy:                  
            pytest.skip()                                                       
        n = 10                                                                  
        k = 3                                                                   
        alpha = 1                                                               
        G = f(n, k, alpha)                                                      
        assert all(d == k for v, d in G.out_degree())                           
        G = f(n, k, alpha, seed=42)                                             
        assert all(d == k for v, d in G.out_degree())
```

... but I leave it up to you - the current setup is completely fine, just wanted to toss a suggestion out there!

Reply:
This alternative seems better imo

Reply:
Agreed!